### ISSUE TITLE
0.3.0 update conflict with GCode Viewer

### ISSUE THREAD

**Issue by khosrua** [kanocz/octopi_eta_override#16]
Since the update, the GCode viewer no longer syncs up with the print properly. It remains static, several layers above the current progress, and refreshes intermittently. The issue is currently managed by disabling the ETA override plugin.

**Comment by kanocz** [comment]
yes, looks like they use the same function override :( so either one or another

**Comment by arekm** [comment]
This is a problem with how OctoPrint perceives "progress". gcodeviewer assumes progress is a point in gcode file not point in printing time.

It could be fixed in octoprint with approach like:
```
diff --git a/src/octoprint/plugins/gcodeviewer/static/js/gcodeviewer.js b/src/octoprint/plugins/gcodeviewer/static/js/gcodeviewer.js
index aaf38a58d..cad450809 100644
--- a/src/octoprint/plugins/gcodeviewer/static/js/gcodeviewer.js
+++ b/src/octoprint/plugins/gcodeviewer/static/js/gcodeviewer.js
@@ -625,7 +625,7 @@ $(function () {
                     self.renderer_syncProgress() &&
                     !self.waitForApproval()
                 ) {
-                    self._renderPercentage(data.progress.completion);
+                    self._renderPercentage(data.progress.fileposcompletion);
                 }
                 self.errorCount = 0;
             } else {
diff --git a/src/octoprint/printer/standard.py b/src/octoprint/printer/standard.py
index 361752645..95ddf0f22 100644
--- a/src/octoprint/printer/standard.py
+++ b/src/octoprint/printer/standard.py
@@ -1113,6 +1113,7 @@ class Printer(PrinterInterface, comm.MachineComPrintCallback):
         else:
             progress = self._comm.getPrintProgress()
             filepos = self._comm.getPrintFilepos()
+            fileposprogress = self._comm.getPrintFileposProgress()
             printTime = self._comm.getPrintTime()
             cleanedPrintTime = self._comm.getCleanedPrintTime()
 
@@ -1164,6 +1165,7 @@ class Printer(PrinterInterface, comm.MachineComPrintCallback):
         return self._dict(
             completion=progress * 100 if progress is not None else None,
             filepos=filepos,
+            fileposcompletion=fileposprogress * 100 if fileposprogress is not None else None,
             printTime=int(printTime) if printTime is not None else None,
             printTimeLeft=int(printTimeLeft) if printTimeLeft is not None else None,
             printTimeLeftOrigin=printTimeLeftOrigin,
diff --git a/src/octoprint/util/comm.py b/src/octoprint/util/comm.py
index 58d9f269a..bc8dcc982 100644
--- a/src/octoprint/util/comm.py
+++ b/src/octoprint/util/comm.py
@@ -1054,6 +1054,11 @@ class MachineCom:
             return None
         return self._currentFile.getProgress()
 
+    def getPrintFileposProgress(self):
+        if self._currentFile is None:
+            return None
+        return self._currentFile.getProgress()
+
     def getPrintFilepos(self):
         if self._currentFile is None:
             return None

```

and then this plugin could only override lower method - ```self._printer._comm.getPrintProgress``` instead.

That would get good UI percentage being shown and also gcodeviewer showing correct action.

Plugin could also stop providing correct progress estimate to octoprint when printing over usb and only provide time (which isn't great solution but ugly workaround for current limitations).

**Comment by arekm** [comment]
@khosrua Could you try 
https://github.com/arekm/octopi_eta_override/archive/devel.zip
(install from octoprint plugin manager; you can just point to URL there)

The new approach is to *only* update progress bar from javascript and never alter OctoPrint python internals (due to OctoPrint assumption mentioned earlier).

**Comment by arekm** [comment]
Fixed in 0.4.0.
